pkg/roachprod/prometheus/prometheus_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package prometheus
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/stretchr/testify/require"
+)
+
+type clusterSpec struct {
+	nodes install.Nodes
+	ips   []string
+}
+
+func TestMakeYAMLConfig(t *testing.T) {
+	testCases := []struct {
+		desc                  string
+		useWorkloadHelpers    bool
+		cluster               *clusterSpec
+		workloadScrapeConfigs []ScrapeConfig
+		expected              string
+	}{
+		{
+			desc:               "multiple scrape nodes",
+			useWorkloadHelpers: false,
+			workloadScrapeConfigs: []ScrapeConfig{
+				{
+					JobName:     "workload0",
+					MetricsPath: "/b",
+					ScrapeNodes: []ScrapeNode{
+						{
+							Nodes: install.Nodes{1},
+							IPs:   []string{"127.0.0.1"},
```

Comment:
@tbg could you explain the purpose of this configuration? Why does a single workload need several scrapeNodes, and within one of these ScrapeNodes, there can be several Nodes?

Reply:
There's no purpose, it's just a unit test. But a workload could be running on multiple nodes, for example a multi-region setup might have a "workload" (TPCC) use partitioning and so there would be multiple instances of the load generator simulating geographically distributed clients.

We can also have multiple workloads running. For example, a cluster might have a kv0 workload running against it, and TPCC as well.

Reply:
Cool. I've simplified the structs so a scrapeNode only represents a single node.